### DISCUSSION TITLE
CSLA 6 save lose data of one field

### DISCUSSION THREAD

**Discussion by Mikael-Pothier** [MarimerLLC/csla#2912] [Questions] [answered]
Hi,

We are trying to update our library to CSLA 6 and we ran into a problem when testing the CRUD operation in a console app. When we do the save operation one of the class properties become empty. All the other properties are fine.

Here is the field in my class, my update method and the code of the console app (I removed all the other field for the example since it didn't seem useful)
![image](https://user-images.githubusercontent.com/8630076/165769100-cda75734-1837-4fce-811d-d7b9238ba399.png)

![image](https://user-images.githubusercontent.com/8630076/165772592-7667d9f3-e2c2-4fe9-90d8-5ec3ed7da815.png)

![image](https://user-images.githubusercontent.com/8630076/165770383-e9b3cc34-cdec-448c-a8bd-aeed4acb8183.png)

This is what happening in Debug. The first image is before the save when my field has data in it and the second is when I get into my update method and the field is empty.

![image](https://user-images.githubusercontent.com/8630076/165773619-c368741f-d483-452e-aecf-2d1db4cd59f0.png)

![image](https://user-images.githubusercontent.com/8630076/165774581-58fb4ffb-1593-4d05-b17e-5d02b631ef43.png)

Also, when I was debugging I found out that when it start, the save goes in the constructor which might be where it get the empty array.
Does anyone have an idea why it loses the date of this field?

**Comment by saulyt** [comment, marked as answer]
GestConcurrence is not declared as a CSLA property, so it is not getting serialized when calling the DataPortal. I'm not sure how it could have worked in a prior version. Perhaps there was no serialization when DataPortal was running local and now there is. Either way, it should be declared as a CSLA property if you are relying on the DataPortal to maintain the state.
```
public static readonly PropertyInfo<byte> GestConcurrenceProperty = RegisterProperty<byte>(nameof(GestConcurrence));
public byte GestConcurrence
{
	get => ReadProperty(GestConcurrenceProperty);
	private set => LoadProperty(GestConcurrenceProperty, value);
}
```

**Reply by Mikael-Pothier** [reply]
Changing for a CSLA property worked perfectly.
Thanks a lot!!

**Reply by rockfordlhotka** [reply]
I'm sure the app was using BinaryFormatter in the past, and that was fine. Microsoft is deprecating BinaryFormatter, and so CSLA now defaults to its own MobileFormatter.

MobileFormatter requires that you used a managed backing field, or if you use a private field you need to override the get/set state methods and manually get/set your private fields into the serialization stream.
